### DISCUSSION TITLE
Redux devtools displaying only one of the two stores' states when it's used with Next.js. Any solution?

### DISCUSSION THREAD

**Discussion by algoORgoal** [pmndrs/zustand#2529] [Q&A] [answered]
## Description

```
<AuthStoreProvider>
    <BoothStoreProvider>
              
    </BoothStoreProvider>
</AuthStoreProvider>
```

- I followed [the official docs on how to set up zustand with next.js](https://docs.pmnd.rs/zustand/guides/nextjs). There are two stores created in the same way.
- When I check the redux devtools, I could only see only of the two store states. Is this behavior intentional? I would like to check both stores at the same time.

### When `<BoothStoreProvider/>` is wrapping around `<AuthStoreProvider/>`

```tsx
<BoothStoreProvider>
  <AuthStoreProvider>
    <div className="mx-auto flex min-h-screen flex-col items-stretch justify-start sm:w-[640px]">
      <Header />
      <div className="flex flex-auto flex-col items-stretch justify-start px-5">
        {children}
      </div>
    </div>
  </AuthStoreProvider>
</BoothStoreProvider>
```

### Result 
![Redux Devtools content](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/rh5jb0vn340c4bkzwtc9.png)
Only the state provided by `<BoothStoreProvider/>` is shown.

### When `<AuthStoreProvider/>` is wrapping around `<BoothStoreProvider/>`

```tsx
<AuthStoreProvider>
  <BoothStoreProvider>
    <div className="mx-auto flex min-h-screen flex-col items-stretch justify-start sm:w-[640px]">
      <Header />
      <div className="flex flex-auto flex-col items-stretch justify-start px-5">
        {children}
      </div>
    </div>
 </BoothStoreProvider>
</AuthStoreProvider>
```

### Result
![Redux Devtools content](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/3o1kixe3k6sjpfadgmvx.png)
Only the state provided by `<AuthStoreProvider/>` is shown.

**Comment by dbritto-dev** [comment]
@algoORgoal would you mind sharing us a minimal repro on stackblitz?

**Reply by algoORgoal** [reply, marked as answer]
@dbritto-dev Thank you for trying to help, but I found out at the moment.
For someone who is new to devtools like me: If you click instance right next to your website title, there should be the rest of the state you defined!
![redux devtools interface](https://github.com/pmndrs/zustand/assets/19303532/e1a200ce-a7d4-4768-b8f6-bfd29c204e9b)

**Comment by carlos-lopez-SaRa** [comment]
The question here is if there's a way to name the instance...

**Reply by dbritto-dev** [reply]
@carlos-lopez-SaRa yes, you can

**Reply by carlos-lopez-SaRa** [reply]
How? it's great that we can, but have yet to find how, didn't see it in the docs...

**Reply by dbritto-dev** [reply]
@carlos-lopez-SaRa devtools has two options: `name` and `store`. The `name` will use for the connection and `store` will be use for the store. Ref: https://github.com/pmndrs/zustand?tab=readme-ov-file#redux-devtools

**Reply by carlos-lopez-SaRa** [reply]
My bad, I already had `{ name: '...' }` on the persist middleware, so I missed the same for the dev tools middleware.
